enhancements/ignition-spec-dual-support.md

```diff
@@ -0,0 +1,327 @@
+ - The user is able to recover the cluster and finish the update if the untranslatable configs are manually translated or remove, or roll back to the old version
+ - The user should have received notification that the update will be changing spec version, as well as received necessary documentation on how to recover a failed update
+ - CI tests are put in place to make sure the existing versions can be updated to the new payload
+
+** As a user of openshift, I’d like to install from a spec 2 bootimage andimmediately update to a spec 3 payload **
+
+Acceptance criteria:
+ - Essentially the same as story 1
+
+** As a user of Openshift, I’d like to install a fresh ignition spec 3 cluster **
+
+Acceptance criteria:
+ - The workflow remains the same for an IPI cluster
+ - The workflow remains the same for a UPI cluster, minus custom specification changes
+ - The user should have good documentation, based on version, of how to set up user defined configs during install time
+
+** As an admin of an existing spec 3 cluster, I’d like to apply a new machineconfig **
+
+Acceptance criteria:
+ - The machineconfig is applied successfully, if the user has defined a correct spec 3 ignition snippet
+ - The user is properly alerted if they attempt to apply a spec 2 config, and the machineconfig fails to apply
+ - The user is given necessary docs to remove the undesired spec 2 config and to translate it to a spec 3 config
+
+** As an admin of an existing spec 3 cluster, I’d like to autoscale a new node **
+
+Acceptance criteria:
+ - The MCS can correct detect which ignition version to serve the bootimage
+ - The bootimage boots correctly and pivots to align correctly to the rest of the cluster version
+
+
+### Risks and Mitigations
+
+** Failing to update a cluster **
+
+The IPI configuration is fully translatable. UPI as well as user provided
+configuration as day 2 operations are not workflows we can guarantee. For
+some users they will simply fail to update the cluster to a new version. To
+mitigate, we must allow the user to recover and/or reconcile, or at the bare
+minimum have comprehensive documentation on what to do in this situation
+
+** Failing to apply a spec 2 machineconfig that worked prior to the update **
+
+Users will likely be unhappy that there is such a large breaking change. In
+other similar cases, e.g. for auto-deployed metal clusters, the served ignition
+configs must all be updated after a certain point of the bootimage to be able
+to bring up new machines. To mitigate we should communicate this change well in
+advance, and provide methods to translate ignition configs. Failed
+installation/alerting systems must clearly communicate the source of error in
+this case, as well as how to mitigate.
+
+** How to properly detect which ignition version to serve new machines **
+
+This will likely be somewhat difficult to handle, as the wrong version on a 
```

Comment:
Can we teach Ignition to request a version using `Accept` headers?  Looks like it may [already do that][1]?

[1]: https://github.com/coreos/ignition/blob/15b5b04ef0c048eebd1cfe5b6192d7a6fc19d625/internal/resource/url.go#L57

Reply:
And with the machine-config server switching on `application/vnd.coreos.ignition+json;version=...` and an ability to translate spec-2 configs to spec-3 and spec-3 configs to spec-2, we'd be able to run a full 4.4.z (or whatever z stream) supporting both configs, and warning about 4.5 (or whatever) incompatibility if we saw any spec-2 requests come in.